### ISSUE TITLE
Can't publish new spring-boot packages referencing the being-published beeline version.

### ISSUE THREAD

**Issue by robbkidd** [honeycombio/beeline-java#173]
Attempting to update the `beelineVersion` referenced by the spring-boot-sleuth-starter and spring-boot-starter packages to the version of the beeline being release causes the release to [fail the publish to Maven](https://app.circleci.com/pipelines/github/honeycombio/beeline-java/966/workflows/5776ebeb-8562-47fb-922f-7ce07a4c3500/jobs/3324). In the first run at releasing 1.7.0:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Beeline Java (Parent) 1.7.0:
[INFO] 
[INFO] Beeline Java (Parent) .............................. SUCCESS [  5.763 s]
[INFO] Beeline Java (Core) ................................ SUCCESS [  0.953 s]
[INFO] Beeline Java (Spring Boot Starter) ................. FAILURE [  1.697 s]
[INFO] Beeline Java (Spring Boot Sleuth Starter) .......... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.368 s
[INFO] Finished at: 2021-12-23T23:30:49Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project beeline-spring-boot-starter: Could not resolve dependencies for project
        io.*********.beeline:beeline-spring-boot-starter:jar:1.7.0: Could not find artifact
        io.*********.beeline:beeline-core:jar:1.7.0 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

**Comment by JamieDanielson** [comment]
Looks like this has likely never worked. The `publish_maven` step has passed on previous tags where `beelineVersion` was left untouched in starter poms, and failed on tags where `beelineVersion` was updated.

**Comment by JamieDanielson** [comment]
Submitted a PR for maven-orb to use a newer maven dependency plugin [https://github.com/CircleCI-Public/maven-orb/pull/24](https://github.com/CircleCI-Public/maven-orb/pull/24). Once that is merged in we can use a newer version of the orb in the Circle config (currently at `circleci/maven@1.0.3`) and this should then work going forward.

If for any reason we need to publish beeline before a new orb can be used, we may also be able to add `mvn compile` into the Circle steps as referenced in this issue [https://github.com/CircleCI-Public/maven-orb/issues/4](https://github.com/CircleCI-Public/maven-orb/issues/4)

**Comment by JamieDanielson** [comment]
PR merged in and new version available https://github.com/CircleCI-Public/maven-orb/pull/24#issuecomment-1016562501
> BotComment: Production version of orb available for use - circleci/maven@1.3.0

next step: try out the new maven orb
